mesonbuild/backend/ninjabackend.py

```diff
@@ -2456,6 +2456,72 @@ def get_link_whole_args(self, linker, target):
         target_args = self.build_target_link_arguments(linker, target.link_whole_targets)
         return linker.get_link_whole_for(target_args) if len(target_args) else []
 
+    def guess_library_absolute_path(self, libname, search_dirs, prefixes, suffixes):
+        for directory in search_dirs:
+            for suffix in suffixes:
+                for prefix in prefixes:
+                    trial = os.path.join(directory, prefix + libname + '.' + suffix)
+                    if os.path.isfile(trial):
+                        return trial
+
+    def guess_external_link_dependencies(self, linker, target, commands, internal):
+        # Ideally the linker would generate dependency information that could be used.
+        # But that has 2 problems:
+        # * currently ld can not create dependency information in a way that ninja can use:
+        #   https://sourceware.org/bugzilla/show_bug.cgi?id=22843
+        # * Meson optimizes libraries from the same build using the symbol extractor.
+        #   Just letting ninja use ld generated dependencies would undo this optimization.
+        search_dirs = []
+        libs = []
+        absolute_libs = []
+
+        build_dir = self.environment.get_build_dir()
+        # the following loop sometimes consumes two items from command in one pass
+        it = iter(commands)
+        for item in it:
+            if item in internal:
+                continue
```

Comment:
Won't this skip if `item` is `-L` and that's already in `internal`  as `['-L', 'foo']` and then cause the library path in the next item to be dropped?

Reply:
`internal` should never contain any options. It's generated by meson for internal dependencies (i.e. output files of targets). And those always are pathes.
But i'll add an additional condition so this only happens when `not item.startswith('-')` as the compiler driver will not use argument starting with dash as direct inputs anyway.